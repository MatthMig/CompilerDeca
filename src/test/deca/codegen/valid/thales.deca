// Description:
// Dans cet exemple, nous utilisons les variables a1, a2, b1, b2, c1 et c2 pour représenter les longueurs de différents segments.
// Ensuite, nous vérifions si les longueurs sont proportionnelles en utilisant l'expression (a1 / a2 == b1 / b2) && (b1 / b2 == c1 / c2).
// Cette expression évalue si le rapport entre les longueurs a1 et a2 est égal au rapport entre les longueurs b1 et b2,
// et si ce dernier rapport est égal au rapport entre les longueurs c1 et c2.

// En fonction du résultat, nous affichons un message approprié en utilisant une instruction if-else.
// Si thales est évalué à true, nous affichons "Les longueurs sont proportionnelles selon le théorème de Thalès".
// Sinon, nous affichons "Les longueurs ne sont pas proportionnelles selon le théorème de Thalès"

{    
    double a1 = 3.0;
    double a2 = 5.0;
    double b1 = 2.0;
    double b2 = 4.0;
    double c1 = 6.0;
    double c2 = 10.0;

    // Vérification du théorème de Thalès
    boolean thales = (a1 / a2 == b1 / b2) && (b1 / b2 == c1 / c2);

    if (thales) {
        println("Les longueurs sont proportionnelles selon le théorème de Thalès.");
    } else {
        println("Les longueurs ne sont pas proportionnelles selon le théorème de Thalès.");
    }
}