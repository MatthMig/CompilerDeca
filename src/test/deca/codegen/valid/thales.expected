Les longueurs ne sont pas proportionnelles selon le théorème de Thalès.